### ISSUE TITLE
Slug not loaded properly when creating an object

### ISSUE THREAD

**Issue by dmathieu** [norman/friendly_id#113]
Hello,

Using Rails 3.1.0.rc1 (with Identity Map disabled), I see a problem which did not occur with the exact same application on 3.0.7.

```
ruby-1.9.2-p136 :007 > c.save!
  SQL (27.7ms)  INSERT INTO "categories" ("cached_slug", "created_at", "image", "locale", "parent_id", "posts_count", "title", "updated_at") VALUES (?, ?, ?, ?, ?, ?, ?, ?)  [["cached_slug", nil], ["created_at", Sat, 04 Jun 2011 09:26:00 UTC +00:00], ["image", nil], ["locale", "fr"], ["parent_id", nil], ["posts_count", 0], ["title", "Testing"], ["updated_at", Sat, 04 Jun 2011 09:26:00 UTC +00:00]]
  Slug Load (13.4ms)  SELECT "slugs".* FROM "slugs" WHERE "slugs"."sluggable_id" IS NULL AND "slugs"."sluggable_type" = 'Category' AND "slugs"."name" = 'testing' AND "slugs"."scope" = 'fr' ORDER BY id DESC
   (0.2ms)  SELECT COUNT(*) FROM "slugs" WHERE "slugs"."name" = 'testing' AND "slugs"."scope" = 'fr' AND "slugs"."sluggable_type" = 'Category'
  SQL (0.7ms)  INSERT INTO "slugs" ("created_at", "name", "scope", "sequence", "sluggable_id", "sluggable_type") VALUES (?, ?, ?, ?, ?, ?)  [["created_at", Sat, 04 Jun 2011 09:26:00 UTC +00:00], ["name", "testing"], ["scope", "fr"], ["sequence", 1], ["sluggable_id", 507], ["sluggable_type", "Category"]]
 => true 
ruby-1.9.2-p136 :008 > c.slug
 => nil 
ruby-1.9.2-p136 :009 > c.id
 => 507 
ruby-1.9.2-p136 :010 > c.reload
  Category Load (0.4ms)  SELECT "categories".* FROM "categories" WHERE "categories"."id" = ? LIMIT 1  [["id", 507]]
```

 => #<Category id: 507, title: "Testing", cached_slug: nil, locale: "fr", created_at: "2011-06-04 09:26:00", updated_at: "2011-06-04 09:26:00", parent_id: nil, image: nil, posts_count: 0> 
    ruby-1.9.2-p136 :011 > c.slug
      Slug Load (0.4ms)  SELECT "slugs".\* FROM "slugs" WHERE "slugs"."sluggable_id" = 507 AND "slugs"."sluggable_type" = 'Category' ORDER BY id DESC LIMIT 1
     => #<Slug id: 1522, name: "testing", sluggable_id: 507, sequence: 1, sluggable_type: "Category", scope: "fr", created_at: "2011-06-04 09:26:00"> 

Basically, the following happens :
I create a new object. Then I try to access it's slug, it's nil.
I reload my object, I can properly access the slug.
